pkg/client/core/service_domain_client.go

```diff
@@ -186,6 +186,7 @@ func (s *domainService) CreatePrimaryHostedZoneWithoutUserinput(_ context.Contex
 		}
 	}
 
+	// Should be done in a Validate() and assumed correct here
```

Comment:
So why do you still check them?

Reply:
It's in my todo, just wanted open a PR while ironing out the minor things